### PR TITLE
Set up associations for App entities

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Created a class `Classroom`
- Created a `has-many/belongs to relationship between `Classroom `and `Student `classes
- Implemented method to ensure that when adding a student to a classroom it also sets the classroom for the student and when setting the classroom for a student it also adds it to the classrooms' students
- Created a class `Book`
- Created a class `Rental`
- Created a `many-to-many` relationship between `Person `and `Book `using the intermediate class `Rental`

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -7,9 +7,10 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (alexander16108) [inline person.rb:15]: - [ ] It required you to create a method for adding a rental with the person sent as a parameter to create a new rental achieving a has-many association 😄

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+  attr_accessor :title, :author
+  attr_reader :rentals
+end
```
Comment (alexander16108) [inline book.rb]: - [ ]  It required you to create a method for adding a rental with the person sent as a parameter to create a new rental achieving a has-many association 😄

**Review by alexander16108 — APPROVED** [review]
Hi @JonahKayizzi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (JonahKayizzi) [reply person.rb:15]: Added. Thanks for the feedback

File: book.rb (hunk 2 above)
Comment (JonahKayizzi) [reply book.rb]: Added. Thanks for the feedback

**Review by omarsalem7 — APPROVED** [review]
Hi @JonahKayizzi 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**